### PR TITLE
show unread messages count for desktop

### LINKED ISSUE #6749: Show unread count instead of 9+
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a chat user, I want to see the real unread message count so that I can know how many unread messages there are.

### Description

Status currently shows unread message counts as `9+` when there are more than 9 unread messages. In common usage this means that chats always tend to show 9+ and offer no context or insight into how many messages are actually unread.  It could be 10 or 945.

By showing the true count we can better inform users to how active a chat is and help them understand how many unread without the need to go into each chat. 

#### Expected behavior
Chat badge count shows actual number if more than 9

#### Actual behavior
Shows 9+

### Solution

For both mobile and desktop.

Discussion: We may want to show 999+ or 9999+ or some upper limit otherwise it could go a bit nuts?

cc @errorists

### PR DESCRIPTION
**Pull request by rcullito**
### Summary:

fixes #6749 

Shows unread messages count for desktop.  

As is suggested in the issue by @errorists, this PR shifts the barrier from "large messages" from anything over 9 to a system that displays the actual count for anything less than 1,000, rounds to the "lowest" thousand for anything up to 10,000, and simply displays "10K+" for anything greater than that. 

The default styling of "large" numbers looked ok on my machine so I left it as is, although comments are welcome of specific scenarios where the current styling does not hold up.

This feature is a case of different implementations for mobile and desktop and this PR specifically addresses the desktop version. I'll open another one for mobile once I've gotten my emulator set up but at least wanted to get this out there for review, comment, and testing to see if others find this schema for displaying messages agreeable. 

One aside for devs reviewing this PR is a question around the preferred method of running the test suite? It was ambiguous to me whether we are using lein for this or clj, so any background here from the last few months would be appreciated! Thanks!

### CODE REVIEW

**Review by dmitryn — APPROVED** [review]

**Review by rasom — APPROVED** [review]

**Comment by chadyj** [comment]
:tada:
Is the plan to do this on mobile too?

**Comment by yenda** [comment]
@rcullito can you make screenshots? Regarding the test suite I still run `lein test-cljs`

**Comment by rcullito** [comment]
@chadyj yes! planning on following up with a mobile PR separately. I just wanted to get this out for review as I was already up and running with my desktop env. 

@yenda thanks for the testing info. I'll attach some screenshots.

**Comment by rcullito** [comment]
variouslargeunreadcounts cc @yenda

**Comment by chadyj** [comment]
Thanks for screenshots!  @errorists take a loo :)

**Comment by errorists** [comment]
it's desktop so cc @EugeOrtiz. Personally, I think it looks bad, the badges should follow the mobile design and be aligned on the right below the timestamp and be allowed to stretch their width to fit the number like [here](https://www.figma.com/file/MRYBI5axcTwL4Ft8xAYhFOfy/Desktop-Chat?node-id=377%3A0)

**Comment by EugeOrtiz** [comment]
+1, they look better to the right. [Designs on Figma](https://www.figma.com/file/MRYBI5axcTwL4Ft8xAYhFOfy/Desktop-Chat?node-id=377%3A0)

I also think the amounts we are showing are weird. What is the difference between saying 900, +4K or +10K? The idea is to say "hey, you have a lot of unread messages here", but I think we can convey that message with a max limit, e.g. +99. From there, it doesn't make much sense to be literal about the amount of messages you have, and also brings a lot of anxiety, because you'll never get to read those 10k messages anyway. 

@rcullito Can we limit the amount and move them to the right?

**Comment by SlizkovaGanna** [comment]
@rcullito  There is an issue with the counter not working properly on Linux.
When having multiple chats, counter is not calculated properly.

image

**Steps to reproduce:**
1. On Linux, login with an existing account with many chats so that counter would be of some quantity already
2. Add a couple of chats, read one of the chats with messages, receive messages in 1-1 and public chats
3. Check that the counter gets updated properly

**Current result:** The counter is updated randomly it seems, not accordingly to messages received/read 
Note: Works perfectly on MacOs

Tested with different scenarios for the counter:

- updating the counter at sending messages/opening chat
- counter updating after fetching messages
- clearing history of chats with old unread messages
- deleting a chat with old unread messages 
- offline/online
- accounts with different amount of messages

Tested on Mac and Linux

**Comment by rcullito** [comment]
Hey @EugeOrtiz and @errorists thank you both for the design feedback! The align-right and stretch-width changes both make sense so I'll aim to match the implementation as close to the Figma images as possible. 

In terms of the actual numbering scheme (99+ versus 4K, etc...) I'm happy to change it to either, I've just seen different suggestions put forth so are you both in consensus now that you'd like to keep it simple at 99+?

**Comment by rcullito** [comment]
@SlizkovaGanna thanks for looking into it across platforms! I'll retest on my local (also linux) to try and diagnose where the counter is deviating from the actual message count.

**Comment by chadyj** [comment]
> In terms of the actual numbering scheme (99+ versus 4K, etc...) I'm happy to change it to either, I've just seen different suggestions put forth so are you both in consensus now that you'd like to keep it simple at 99+?

For numbering I'd be happy to see it stay as-is.  I've often joined telegram groups after a hiatus to see huge numbers of unread messages.  It is unlikely we will ever see huge numbers in Status since we only fetches max 24 hours (at least currently) so the numbers will be manageable and may as well report real numbers.

>  it doesn't make much sense to be literal about the amount of messages you have, and also brings a lot of anxiety, because you'll never get to read those 10k messages anyway.

Mark as read coming soon! 🙏

**Comment by rcullito** [comment]
updatedstylingsamenumbernomorehomecount

thanks @chadyj for clarifying! here's a snapshot of the sidebar with a similar numbering to yesterday but updated styling

**Comment by rcullito** [comment]
@SlizkovaGanna in looking at the latest designs on FIgma I actually saw that the counter is not even present on the home tab, so I've since removed it and (conveniently enough right??) this hopefully prevents us from having to address the mismatch in count issue that you found since that component will be going away entirely. 

Let me know if there are any other discrepancies between the PR and what you were expecting, thanks!

**Comment by errorists** [comment]
Thanks @rcullito Another thing which caught my eye, we unblocked the counter next to the 'Home' tab icon, which highlighted the fact it doesn't quite work right. It shouldn't count the total messages but rather show the number of chats that have new messages. So say you have 3 chats with 200 new messages in them, the Home icon shouldn't show you 200 but rather 3. Can we have that fixed as part of this issue? :)

**Comment by SlizkovaGanna** [comment]
@rcullito I can see the counter on the Home tab with the current build for Mac, it's there :)

image

> looking at the latest designs on FIgma I actually saw that the counter is not even present on the home tab, so I've since removed it

> We unblocked the counter next to the 'Home' tab icon, which highlighted the fact it doesn't quite work right. It shouldn't count the total messages but rather show the number of chats that have new messages. So say you have 3 chats with 200 new messages in them, the Home icon shouldn't show you 200 but rather 3. Can we have that fixed as part of this issue? :)

@errorists @chadyj @EugeOrtiz We need to have a final decision on the counter for Home tab :)
It's not present on the design page - https://www.figma.com/file/MRYBI5axcTwL4Ft8xAYhFOfy/Desktop-Chat?node-id=377%3A0 , as @rcullito pointed out - should it exist? If so, should we add the functional logic @errorists  described?

Also, a **separate** question on that topic - the icon counter we have for the app, currently counts only messages received in 1-1 and group chats, not including public chats.
So, saying user has 1 unread message in 1-1 chat, 1 in a public chat and 1 in a group chat, the icon counter will only show number 2 (1-1 + group chat), and not 3:

monosnap 2018-11-29 10-51-16

@errorists @EugeOrtiz @chadyj should we imply the same logic as for the Home tab counter here? (I will create a separate issue for this with your confirmation)

<blockquote><div> Figma</div><div><strong><a href="https://www.figma.com/file/MRYBI5axcTwL4Ft8xAYhFOfy/Desktop-Chat">Figma</a></strong></div><div>Created with Figma</div></blockquote>

**Comment by SlizkovaGanna** [comment]
Moving back to Contributor column, since there seems to be a change of logic.
Waiting for the main question in the comment https://github.com/status-im/status-react/pull/6888#issuecomment-442778078 to be answered (and also hopefully the **separate** question too :)).

**Comment by EugeOrtiz** [comment]
Here is my proposal:
- Badge on icon (OS dock) should indicate the total amount of unread messages, no matter where they come from (channel, group or 1:1 conversations). The idea is that when you are not in the app, we tell you that you have unread messages, even if they are from a public conversation you are following. I created an [issue](https://github.com/status-im/status-react/issues/6672) for this a while ago. Telegram does this.

- Each conversation/channel/group should indicate its own amount of unread messages (what Rob did).

- The home tab should indicate that there is activity going on (see proposed design). I think we will be repeating information by adding again the total amount of unread messages (already on dock). A simple "dot" could be used to indicate that there is activity going on. We could also use this for Profile when we need to get the user's attention. [Design proposal](https://www.figma.com/file/MRYBI5axcTwL4Ft8xAYhFOfy/Desktop-Chat?node-id=377%3A0):
group 19

group 22

Let me know what you think.

**Comment by errorists** [comment]
+1 to what Euge said, makes sense. I'd like to clarify one bit though, with the total unread badge on the dock icon, you mean the sum total number of unread messages across all chats or the number of chats with unread messages, afaik Telegram, Messenger or Whatsapp follow the second option and I'd assume we want to also. Having the number on the dock badge update for each new single message across all chats can be irritating and I feel unnecessary.

**Comment by SlizkovaGanna** [comment]
@EugeOrtiz 
I like your proposal, it seems convenient and logical :)
@errorists I also think you have a good point - I'll copy-paste your comment to -> https://github.com/status-im/status-react/issues/6672 - the ticket @EugeOrtiz created for the dock icon specifically to keep the convo separated.

So, if we have a consensus , we should:
**1)** Remove the counter for Home tab and leave the dot indicating any activity - as in mockups 
**2)** Each conversation should indicate it's own amount of unread messages (already implemented) - as in mockups
Thank you, guys!

**Comment by rcullito** [comment]
@EugeOrtiz would you mind linking to an asset for the home icon "with the dot" ? I was assuming this would be its own image rather than us adding a layer from our side, but let me know what you think, thanks!

**Comment by rcullito** [comment]
updatedchatcountsstoichomeicon

hey @chadyj this PR has gotten a bit stale so I wanted to poll your opinion on going ahead and merging this fix for updated chat counts (which was what the issue was originally raised for) and tackling the home tab design in a separate issue/PR once we have consensus there and a clear plan. Attached is a photo of the current state of counts from my local. Thanks!

**Comment by chadyj** [comment]
+1 to merge, and follow up with home dot for launch (Plus any design tweaks needed from design team). Let's hustle and get it out and clear the deck!

@EugeOrtiz @MaxRis @SlizkovaGanna @errorists @siphiuel @vkjr @churik Leave a 👍 or 👎 on this comment.

**Comment by chadyj** [comment]
+1 to merge, and follow up with home dot for launch (Plus any design tweaks needed from design team). Let's hustle and get it out and clear the deck!

@EugeOrtiz @MaxRis @SlizkovaGanna @errorists @siphiuel @vkjr @churik Leave a 👍 or 👎 on this comment.

**Comment by chadyj** [comment]
@rcullito results:

👍 merge: 6 votes
👎 don't merge: 0 votes

The people have spoken!

**Comment by churik** [comment]
@chadyj let me retest it one more time just in case, OK?

**Comment by chadyj** [comment]
Follow up issue to fix tab badge > https://github.com/status-im/status-react/issues/6672

**Comment by churik** [comment]
@rcullito cannot build it; I see several commits from you after last build https://github.com/status-im/status-react/pull/6888#issuecomment-442601343, but couldn't build new one to test it: https://ci.status.im/job/status-react/job/pull%20requests/view/change-requests/job/PR-6888/

Don't think that problem actually with your PR, but just to start I would ask you to rebase it to current `develop` and take a look into  failures in builds. Thanks!

**Comment by churik** [comment]
@siphiuel can you rebase it please and I'll try to rebuild?

**Comment by vitvly** [comment]
@churik rebased

**Comment by churik** [comment]
all remaining issues are mentioned in https://github.com/status-im/status-react/issues/6672.
PR doesn't affect mobile builds.

**Comment by churik** [comment]
@siphiuel can you please merge it?

**Comment by vitvly** [comment]
@churik @yenda tried to merge via `./scripts/merge-pr.sh 6888`, but got

```
remote: error: GH006: Protected branch update failed for refs/heads/develop.
remote: error: At least 2 approving reviews are required by reviewers with write access.
To github.com:status-im/status-react.git
 ! [remote rejected]     develop -> develop (protected branch hook declined)
error: failed to push some refs to 'git@github.com:status-im/status-react.git'
```
Do you know what could have caused this?

**Comment by vitvly** [comment]
Ah, maybe it's because it's being built after squash.

**Comment by vitvly** [comment]
Merged.

**Pull request closed without merging**
